### DISCUSSION TITLE
pwa's questions for next.js...

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#39831] [Help] [answered]
So I am trying to convert a nextjs app into a pwa..

I have tried 2 different routes:

1. install next-pwa..
2. create a manifest in public folder .. including icons 
3. link the manifest to _app.js
4. setup the next config
5. run build
6. ... and nothing else happens... it suppose to build a few files automatically sw.js , but it isn't doing it.

The other approach is not to use next-pwa...

1. next-pwa already installed.. so just leave it there
2. create a manifest in public folder .. including icons
3. link manifest to _app.js
4.  Leave next-config normal without next-pwa setting
5.  npm run build
6. npm run start
7. check the browser....  still no install button from pwa...

In the both cases above when I inspect the console under Application it shows me the manifest with proper information... and "no matching service workers detected"....   and no install button on the browser for the app..

During the Next-pwa plugin test it is supposed to create all the service workers folders and all, but they are not there...

During the manual setup without next-pwa I haven't gotten to create a service worker because I am not that far.... but shouldn't  be ready for the app to download  with just a manifest and a link to _app.js???

**Comment by Elindo586** [comment]
repo link here:
https://github.com/Elindo586/nextprojectb



Oh... I forgot ..... I also did create a server which it was mentioned in the next-pwa installation..... so that server still there, but I don't think is doing anything...

**Comment by Elindo586** [comment]
I get the feeling that...

1. next-pwa just isn't working...
2. the other manual instructions that I am following might be outdated, and I need to create service workers before I see a download button on the browser for the app..
![Capture333](https://user-images.githubusercontent.com/93617433/185999685-e732a8ce-1815-4656-8041-404663fbad20.PNG)

**Comment by leerob** [comment, marked as answer]
Hey everyone. We now have updated documentation on building PWAs with Next.js.

https://nextjs.org/docs/app/building-your-application/configuring/progressive-web-apps
